syntax = "proto3";

package claims_processing;

import "google/protobuf/empty.proto";

message ClaimsRequest {
  string claimId = 1;
  string x_tenant_id = 2; // New field added
}

message HealthCheckRequest {}

message HealthCheckResponse {
  string status = 1;
}

message LogViewRequest {
  string logType = 1; // Accepts "error" or "info"
}

message LogViewResponse {
   string log_text = 1;
}

service ClaimsProcessing {
  rpc ProcessClaim(ClaimsRequest) returns (google.protobuf.Empty) {}
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse) {}
  rpc ViewLogs(LogViewRequest) returns (LogViewResponse) {} 
}
